### ISSUE TITLE
Data Folders

### ISSUE THREAD

**Issue by rsameera99** [Mustafa-MS/3D-CNN-LUNA16#2]
Hello! Firstly, your repository is very simple and easily to understand for a newbie like me so thank you very much. I am currently experimenting with CNN and LUNA16 dataset. May I know how do you organize the folders for LUNA16 data set? There are subset0 to subset10 right? also there are a few csv files too. How do you manage the paths?

Thank you in advance!

**Comment by Mustafa-MS** [comment]
Hi rsameera99.
I'm glad to help!
I've downloaded the LUNA16 as 10 separate folders from (0 to 9)
each folder have 88 patients.
It was used for 10 folding cross validation during the competition.
by using these lines of code 
```
luna_path = "/home/mustafa/project/LUNA16/"
.
.
.
for subsetindex in range(9):
    luna_subset_path = luna_path + "subset" + str(subsetindex) + "/"
    #luna_subset_path = '/home/mustafa/project/Testfiles/train/'
    file_list = glob.glob(luna_subset_path + '*.mhd')
    for file_path in file_list:
        #print (file_path)
        x_train.append(process_scan(file_path))
        print("xtrain = ", len(x_train))
```
I surf each folder from 0 to 8 and then load it for a training.
then I load the folder #9 for validating
```
luna_test_path = '/home/mustafa/project/LUNA16/subset9/'
test_file_list = glob.glob(luna_test_path + '/*.mhd')
.
.
.
for file_path in test_file_list:
    #print(file_path)
    x_val.append(process_scan(file_path))
    print("x_val = ", len(x_val))
```

**Comment by rsameera99** [comment]
oh I see. Thank you very much for the explanation, your time and response. I really appreciate it.

**Comment by rsameera99** [comment]
Hi, I have another question, what is the function of wandbcallback? How can I edit it to be from my machine? thank you

**Comment by Mustafa-MS** [comment]
Hi !
This is a function to monitor, log and analyze the machine learning experiments, I used it to monitor my GPU usage (for debugging).
You can totally remove it, and the code will works just well.
Or, you can read the [documentation](https://docs.wandb.ai/guides/integrations/keras) about how to use it.

**Comment by baotoann9** [comment]
Hi!!!
The way you do it is very nice I am following your instructions. But I have been training model for hours and epoch is still not working. Can you help me? Thank you very much.
# Train the model, doing validation at the end of each epoch
epochs = 5
model.fit(
    train_dataset,
    validation_data=validation_dataset,
    epochs=epochs,
    shuffle=True,
    verbose=2,
    callbacks=[checkpoint_cb, early_stopping_cb, WandbCallback()],
)
